code/modules/unit_tests/fish_unit_tests.dm

```diff
@@ -280,15 +280,41 @@
 	growth_rate = 100
 	fish_traits = list() //We don't want to end up applying traits twice on the resulting lobstrosity
 
-/datum/unit_test/explosive_fishing
+/datum/unit_test/fish_sources
 
-/datum/unit_test/explosive_fishing/Run()
-	var/datum/fish_source/source = GLOB.preset_fish_sources[/datum/fish_source/unit_test]
+/datum/unit_test/fish_sources/Run()
+	var/datum/fish_source/source = GLOB.preset_fish_sources[/datum/fish_source/unit_test_explosive]
 	source.spawn_reward_from_explosion(run_loc_floor_bottom_left, 1)
 	if(source.fish_counts[/obj/item/wrench])
 		TEST_FAIL("The unit test item wasn't removed/spawned from fish_table during 'spawn_reward_from_explosion'.")
 
-/datum/fish_source/unit_test
+	///From here, we check that the profound_fisher as well as fish source procs for rolling rewards don't fail.
+	source = GLOB.preset_fish_sources[/datum/fish_source/unit_test_profound_fisher]
+	run_loc_floor_bottom_left.AddElement(/datum/element/lazy_fishing_spot, /datum/fish_source/unit_test_profound_fisher)
+	var/mob/living/basic/fisher = allocate(/mob/living/basic)
+	fisher.AddComponent(/datum/component/profound_fisher)
+	fisher.set_combat_mode(FALSE)
+	fisher.melee_attack(run_loc_floor_bottom_left, ignore_cooldown = TRUE)
+	if(source.fish_counts[/obj/item/fish/testdummy] != 1)
+		TEST_FAIL("The unit test profound fisher didn't catch the test fish on a lazy fishing spot (element)")
+	///For good measure, let's try it again, but with the component this time, and a human mob and gloves
+	run_loc_floor_bottom_left.RemoveElement(/datum/element/lazy_fishing_spot, /datum/fish_source/unit_test_profound_fisher)
+	var/datum/component/comp = run_loc_floor_bottom_left.AddComponent(/datum/component/fishing_spot, source)
+	var/mob/living/carbon/human/consistent/angler = allocate(/mob/living/carbon/human/consistent)
+	var/obj/item/clothing/gloves/noodling = allocate(/obj/item/clothing/gloves)
+	noodling.AddComponent(/datum/component/profound_fisher)
+	angler.equip_to_slot(noodling, ITEM_SLOT_GLOVES)
+	angler.UnarmedAttack(run_loc_floor_bottom_left, proximity_flag = TRUE)
+	if(source.fish_counts[/obj/item/fish/testdummy])
```

Comment:
It would really help with readability if you added some empty lines inbetween, like paragraphs between relevant code